File: step-3/be/src/models/Task.ts
import { ObjectId } from "mongodb";

export interface Task {
  name: string;
  assignees: Array<string>;
  currentAssignee?: string;
  id?: ObjectId;
}

export const taskJsonSchema = {
  $jsonSchema: {
    bsonType: "object",
    required: ["name", "assignees", "currentAssignee"],
    additionalProperties: false,
    properties: {
      name: { bsonType: "string" },
      assignees: { bsonType: "array" },
      currentAssignee: { bsonType: "string" },
    },
  },
};

File: step-3/be/types/task.go
package types

type Task struct {
	Name            string   `bson:"name"`
	Assignees       []string `bson:"assignees"`
	CurrentAssignee string   `bson:"currentAssignee"`
	ID              string   `bson:"_id,omitempty"`
}

func ValidateTask(task *Task) bool {
	return task.Name != "" && task.ID != ""
}
